File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.cpp
class Solution {
public:
    string removeDuplicates(string s) {
        string result;
        for (int index = 0; index < s.size(); index ++){
            if (!result.empty() and s[index] == result.back()){
                result.pop_back();
                }   
            
            else{
                result.push_back(s[index]);
            }
            }
        
        return result;
    }
};

File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        stack = []
        for index in range(len(s)):
            if stack and stack[-1] == s[index]:
                stack.pop()
            else:
                stack.append(s[index])
        return "".join(stack)